compute/cloud-client/src/main/java/compute/CreateWithLocalSsd.java

```diff
@@ -0,0 +1,197 @@
+  //    with the VM that uses it.
+  private static AttachedDisk diskFromImage(String diskType, int diskSizeGb, boolean boot,
+      String sourceImage, boolean autoDelete) {
+
+    AttachedDiskInitializeParams attachedDiskInitializeParams =
+        AttachedDiskInitializeParams.newBuilder()
+            .setSourceImage(sourceImage)
+            .setDiskSizeGb(diskSizeGb)
+            .setDiskType(diskType)
+            .build();
+
+    AttachedDisk bootDisk = AttachedDisk.newBuilder()
+        .setInitializeParams(attachedDiskInitializeParams)
+        // Remember to set auto_delete to True if you want the disk to be deleted when you delete
+        // your VM instance.
+        .setAutoDelete(autoDelete)
+        .setBoot(boot)
+        .build();
+
+    return bootDisk;
+  }
+
+  // Create an AttachedDisk object to be used in VM instance creation. The created disk contains
+  // no data and requires formatting before it can be used.
+  // Args:
+  //    zone: The zone in which the local SSD drive will be attached.
+  private static AttachedDisk localSsdDisk(String zone) {
+
+    AttachedDiskInitializeParams attachedDiskInitializeParams =
+        AttachedDiskInitializeParams.newBuilder()
+            .setDiskType(String.format("zones/%s/diskTypes/local-ssd", zone))
+            .build();
+
+    AttachedDisk disk = AttachedDisk.newBuilder()
+        .setType(AttachedDisk.Type.SCRATCH.name())
+        .setInitializeParams(attachedDiskInitializeParams)
+        .setAutoDelete(true)
+        .build();
+
+    return disk;
+  }
+
+  // Send an instance creation request to the Compute Engine API and wait for it to complete.
+  // Args:
+  //    project: project ID or project number of the Cloud project you want to use.
+  //    zone: name of the zone to create the instance in. For example: "us-west3-b"
+  //    instanceName: name of the new virtual machine (VM) instance.
+  //    disks: a list of compute.v1.AttachedDisk objects describing the disks
+  //           you want to attach to your new instance.
+  private static Instance createInstance(String project, String zone, String instanceName,
+      List<AttachedDisk> disks)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `instancesClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (InstancesClient instancesClient = InstancesClient.create()) {
+
+      // machineType: machine type of the VM being created. This value uses the
+      // following format: "zones/{zone}/machineTypes/{type_name}".
+      // For example: "zones/europe-west3-c/machineTypes/f1-micro"
+      String machineType = String.format("zones/%s/machineTypes/%s", zone, "n1-standard-1");
```

Comment:
`n1-standard-1` should be replaced as a variable or inlined into the string itself.